test/hotspot/jtreg/compiler/cha/StrengthReduceInterfaceCall.java

```diff
@@ -32,13 +32,15 @@
  * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
  *
  * @run main/othervm -Xbootclasspath/a:. -XX:+IgnoreUnrecognizedVMOptions -XX:+UnlockDiagnosticVMOptions
+ *                   -XX:+UseVtableBasedCHA
```

Comment:
`UseVtableBasedCHA` is `true` by default. May be duplicate these `runs` for both cases: on and off.

Reply:
The test assumes `UseVtableBasedCHA` is on and some of the assertions it makes fail with `-XX:-UseVtableBasedCHA`. So, it would require all the changes in the test logic have to be guarded by a check whether UseVtableBasedCHA is on or off.

Considering I propose to have `UseVtableBasedCHA` turned on by default, I don't see much value in complicating the test.

Reply:
So only new implementation is tested with these changes in test. Got it. May be use:
`@requires vm.opt.final.UseVtableBasedCHA == true`

Reply:
I'm fine with both approaches. 

Explicitly setting the flag looked to me more robust and clearer communicating the intent. But if you prefer `@requires`, I'll use it.

Reply:
Let hear @iignatev opinion.

Reply:
from my point of view, `@requires` is clearer and also eliminates "wasted" execution (if someone tries to run this test w/ `-XX:-UseVtableBasedCHA`), so I'd prefer if we use it.

I have a more generic comment about `UseVtableBasedCHA`. I understand the desire to introduce a flag to switch back to the old implementation, but I'm somewhat concern that it adds a new dimension into configuration space that won't be covered by our existing tests (w/ the test which exercises interesting parts of the related code is inapplicable) and isn't part of our regular test configurations. Can we make it an experimental flag (w/ vtable-based CHA still being enabled by default)? this way, the quality bar for the old implementation will be somewhat lower, yet the end-users will still be able to return to the old implementation if it, for some reason, works better in their use-cases.

-- Igor